Samples/PowerShell/ADLSUser/Set-AdlsAccess.ps1

```diff
@@ -134,49 +185,81 @@ function copyacls
-try
-{   if($EntityType -ieq "other" -and !([string]::IsNullOrEmpty($EntityIdToAdd)))
-	{
-		throw "EntityIdToAdd is not supported when modifying permissions for 'Other' entity types"
-	}
-	
-	if($EntityType -ine "other" -and [string]::IsNullOrEmpty($EntityIdToAdd))
-	{
-		throw "EntityIdToAdd is required when modifying permissions for 'user' and 'group' entity types"
-	}
+
+switch ($PsCmdlet.ParameterSetName) 
+{ 
+    “SetOwner”  {
+        try
+        {
+            if($EntityType -ieq "other")
+	        {
+		        throw "SetOwner is not supported when 'Other' EntityType is specified"
+	        }
 	
-	$ignored = [Guid]::Empty
-	if($EntityType -ine "other" -and !([Guid]::TryParse($EntityIdToAdd, [ref]$ignored)))
-	{
-		throw "EntityIdToAdd must be a valid GUID. EntityIdToAdd value was: $EntityIdToAdd"
-	}
+	        if($EntityType -ine "other" -and [string]::IsNullOrEmpty($EntityId))
+	        {
+		        throw "EntityId is required when SetOwner is specified"
+	        }
 	
-    $executingDir = Split-Path -parent $MyInvocation.MyCommand.Definition
-    $executingFile = Split-Path -Leaf $MyInvocation.MyCommand.Definition
-    
-    $profilePath = Join-Path $env:TEMP "jobprofilesession.tmp"
-    if(Test-Path $profilePath)
-    {
-        Remove-Item $profilePath -Force -Confirm:$false
-    }
+	        $ignored = [Guid]::Empty
+	        if($EntityType -ine "other" -and !([Guid]::TryParse($EntityId, [ref]$ignored)))
+	        {
+		        throw "EntityId must be a valid GUID. EntityId value was: $EntityId"
+	        }
     
-    Save-AzureRMProfile -path $profilePath | Out-Null
+            Write-Host "Request to change owner to entity $EntityId successfully submitted and will propagate over time depending on the size of the folder."
+            Write-Host "Please do not close this PowerShell window; otherwise, the propagation will be cancelled."
+            
+            Set-AzureRmDataLakeStoreItemOwner -Account $Account -Path $Path -Type $EntityType -Id $EntityId -WarningAction SilentlyContinue | Out-Null
+
+            setownerrec -Account $Account -Path $Path -Id $EntityId -EntityType $EntityType -WarningAction SilentlyContinue | Out-Null
+
+            Write-Host "Done."
+        }
+        catch
+        {
+            Write-Error "Set owner operation failed with the following error: $($error[0])"
+        }
+        break
+    } 
+    “SetAclEntry”  {
+        try
+        {
+            if($EntityType -ieq "other" -and !([string]::IsNullOrEmpty($EntityId)))
+	        {
+		        throw "EntityId is not supported when modifying permissions for 'Other' entity types"
+	        }
+	
+	        if($EntityType -ine "other" -and [string]::IsNullOrEmpty($EntityId))
+	        {
+		        throw "EntityId is required when modifying permissions for 'User' and 'Group' entity types"
+	        }
+	
+	        $ignored = [Guid]::Empty
+	        if($EntityType -ine "other" -and !([Guid]::TryParse($EntityId, [ref]$ignored)))
```

Comment:
Can't we make entityId be a guid as a parameter? Then you don't need to try and parse it.